### PR TITLE
fix(textbox): fix textbox losing focus on WASM if alt-tabbed out then back in

### LINKED ISSUE #14965: [WASM] Suggestion is not selected when pressing down arrow key in suggestion list of AutoSuggestBox
### Current behavior

This is the AutoSuggestBox issue on WASM.
When typing keyword to text box of AutoSuggestBox, SuggestionList popup will come.
Here, when I press down arrow key once, the first suggestion isn't selected. 
But then when I press up arrow key, last suggestion is selected. 
As a result, the down arrow key doesn't work in suggestion list of AutoSuggestBox

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

[1.zip(updated)](https://github.com/unoplatform/uno/files/13861864/1.zip)

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

5.1.0-dev.1219

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### LINKED ISSUE #14967: [WASM] Suggestion list isn't displayed even if typing the keyword in AutoSuggestBox
### Current behavior

Sometimes, suggestion list never be displayed even if typing the keyword to text box of AutoSuggestBox

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

1. Enter the word to text box of AutoSuggestBox. Suggestion list will be displayed.
2. Press Alt+Tab and then press Alt+Tab again.(on windows)
3. Enter the word to text box again. Suggestion list never be displayed.

https://github.com/unoplatform/uno/files/13861864/1.zip
(updated)
### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

5.1.0-dev.1219

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #14967

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Input/FocusManager.wasm.cs [hunk 1]
```diff
@@ -44,10 +46,30 @@ internal static void ProcessElementFocused(UIElement element)
 				_log.Value.LogDebug($"{nameof(ProcessElementFocused)}() focusedElement={GetFocusedElement()}, element={element}, searching for focusable parent control");
 			}
 
-			foreach (var parent in element.GetParents())
+			foreach (var candidate in element.GetParents().Prepend(element))
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Input/FocusManager.wasm.cs:49]: Consider adding an optional `bool includeSelf` parameter to GetParents that defaults to false.

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Input/FocusManager.wasm.cs:49]: Actually, there is already `GetAllParents` that will do that job

File: src/Uno.UI/UI/Xaml/Input/FocusManager.wasm.cs [hunk 2]
```diff
@@ -44,10 +46,30 @@ internal static void ProcessElementFocused(UIElement element)
...
+					{
+						if (_log.Value.IsEnabled(LogLevel.Debug))
+						{
+							_log.Value.LogDebug(
+								$"Root visual focused - caused by browser keyboard navigation to the page, " +
+								$"moving focus to actual first focusable element - {frameworkElement?.ToString() ?? "[null]"}.");
+						}
+						frameworkElement.Focus(FocusState.Keyboard);
+					}
+					return;
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Input/FocusManager.wasm.cs:70]: Is this `return;` needed?

Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/Input/FocusManager.wasm.cs:70]: Yes. Once we found the right candidate, we should stop. I can probably change it to `break` to match the other if branches

**Comment by TopProgrammer77** [comment]
@ramezgerges , this isn't related to #14965 . This is just #14967 , I think.

**Comment by ramezgerges** [comment]
Yes, well spotted. Fixed.

File: src/Uno.UI/UI/Xaml/Input/FocusManager.wasm.cs [hunk 3]
```diff
@@ -44,10 +47,30 @@ internal static void ProcessElementFocused(UIElement element)
...
+			foreach (var candidate in element.GetAllParents())
 			{
 				// Try to find the first focusable parent and set it as focused, otherwise just keep it for reference (GetFocusedElement())
-				if (parent is TextBlock textBlock && textBlock.IsFocusable)
+
+				// Special handling for RootVisual - which is not focusable on managed side
+				// but is focusable on native side. The purpose of this trick is to allow
+				// us to recognize, that the page was focused by tabbing from the address bar
+				// and focusing the first focusable element on the page instead.
+				if (candidate is RootVisual rootVisual)
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Input/FocusManager.wasm.cs:58]: @ramezgerges Can you try creating a UI test (or runtime test) that validates this particular block? Are we able to programmatically invoke this particular piece?

Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/Input/FocusManager.wasm.cs:58]: I don't think so. This only gets called on `ReceivedFocusNative`. This is preexisting code by the way, I just moved it inside `ProcessElementFocused` where it makes more sense IMHO.

Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/Input/FocusManager.wasm.cs:58]: Actually, after thinking about some hacks, there doesn't seem to be a straightforward way to realisticly unfocus and refocus a tab/window I think it's not that critical and it hasn't been touched in ages, so it's probably fine.

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Input/FocusManager.wasm.cs:58]: Ok, then let's make a sample marked as a manual test which explains what to do and what to expect. Thanks!

**Review by MartinZikmund — APPROVED** [review]

**Pull request merged**